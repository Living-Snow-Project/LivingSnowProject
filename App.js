import React from 'react';
import { StatusBar } from 'expo-status-bar';
import AppLoading from 'expo-app-loading';
import { Asset } from 'expo-asset';
import { PropTypes } from 'prop-types';
import * as Font from 'expo-font';
import * as Icon from '@expo/vector-icons';
import Navigation from './navigation/MainTabNavigator';
import { Storage } from './lib/Storage';
import { SafeAreaProvider } from 'react-native-safe-area-context';

global.appConfig = {
  name: '',
  organization: '',
  showFirstRun: true,
  showGpsWarning: true
}

export default class App extends React.Component {
  static propTypes = {
    skipLoadingScreen: PropTypes.bool,
  }
  
  state = {
    isLoadingComplete: false
  }

  render() {
    if (!this.state.isLoadingComplete && !this.props.skipLoadingScreen) {
      return (
        <AppLoading
          startAsync={this._loadResourcesAsync}
          onError={this._handleLoadingError}
          onFinish={this._handleFinishLoading}
        />
      );
    } else {
      return (
        <SafeAreaProvider>
          <Navigation />
          <StatusBar />
        </SafeAreaProvider>
      );
    }
  }

  _loadResourcesAsync = async () => {
    return Promise.all([
      Asset.loadAsync([
        require('./assets/images/robot-dev.png'),
        require('./assets/images/robot-prod.png'),
      ]),
      Font.loadAsync({
        // This is the font that we are using for our tab bar
        ...Icon.Ionicons.font,
        // We include SpaceMono because we use it in HomeScreen.js. Feel free
        // to remove this if you are not using it in your app
        'space-mono': require('./assets/fonts/SpaceMono-Regular.ttf'),
      }),
      Storage.loadAppConfig()
    ]);
  };

  _handleLoadingError = error => {
    // In this case, you might want to report the error to your error
    // reporting service, for example Sentry
    console.warn(error);
  };

  _handleFinishLoading = () => {
    this.setState({ isLoadingComplete: true });
  };
}
