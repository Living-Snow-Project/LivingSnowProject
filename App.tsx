import React from "react";
import { StatusBar } from "expo-status-bar";
import { SafeAreaProvider } from "react-native-safe-area-context";
import Navigation from "./src/navigation/MainTabNavigator";
import useCachedResources from "./src/hooks/useCachedResources";

export default function App() {
  const isLoadingComplete = useCachedResources();

  if (!isLoadingComplete) {
    return null;
  }

  return (
    <SafeAreaProvider>
      <Navigation />
      <StatusBar />
    </SafeAreaProvider>
  );
}
