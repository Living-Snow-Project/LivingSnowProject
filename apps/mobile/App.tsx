import React from "react";
import { StatusBar } from "expo-status-bar";
import "react-native-gesture-handler";
import NativeBaseProvider from "./src/providers";
import Navigation from "./src/navigation/MainTabNavigator";
import useCachedResources from "./src/hooks/useCachedResources";
import {
  useAlgaeRecords,
  AlgaeRecordsContext,
} from "./src/hooks/useAlgaeRecords";

export default function App() {
  const [algaeRecords] = useAlgaeRecords();
  const isLoadingComplete = useCachedResources();

  if (
    algaeRecords.getCurrentState() !== "Seeding" &&
    !algaeRecords.isSeeded()
  ) {
    algaeRecords.seed();
  }

  if (!isLoadingComplete || !algaeRecords.isSeeded()) {
    return null;
  }

  return (
    <AlgaeRecordsContext.Provider value={algaeRecords}>
      <NativeBaseProvider>
        <Navigation />
        <StatusBar />
      </NativeBaseProvider>
    </AlgaeRecordsContext.Provider>
  );
}
