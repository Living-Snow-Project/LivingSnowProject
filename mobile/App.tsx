import React from "react";
import { StatusBar } from "expo-status-bar";
import { SafeAreaProvider } from "react-native-safe-area-context";
import "react-native-gesture-handler";
import Navigation from "./src/navigation/MainTabNavigator";
import useCachedResources from "./src/hooks/useCachedResources";
import {
  useRecordReducer,
  RecordReducerStateContext,
  RecordReducerActionsContext,
} from "./src/hooks/useRecordReducer";

export default function App() {
  const isLoadingComplete = useCachedResources();
  const [storageState, storageActions] = useRecordReducer();

  if (storageState.state !== "Seeding" && !storageState.seeded) {
    storageActions.seed();
  }

  if (!isLoadingComplete || !storageState.seeded) {
    return null;
  }

  return (
    <RecordReducerStateContext.Provider value={storageState}>
      <RecordReducerActionsContext.Provider value={storageActions}>
        <SafeAreaProvider>
          <Navigation />
          <StatusBar />
        </SafeAreaProvider>
      </RecordReducerActionsContext.Provider>
    </RecordReducerStateContext.Provider>
  );
}
