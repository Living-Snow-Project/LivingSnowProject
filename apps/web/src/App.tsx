import { useEffect, useState } from "react";
import { RecordsApiV2 } from "@livingsnow/network";
import { TableHeader, TableRow } from "./components/TableRow";
import {
  useMsal,
  useIsAuthenticated,
  AuthenticatedTemplate,
  UnauthenticatedTemplate,
} from "@azure/msal-react";

import "./App.css";

function App() {
  const [records, setRecords] = useState<JSX.Element[]>([]);
  const { accounts, instance } = useMsal();
  const isAuthenticated = useIsAuthenticated();

  useEffect(() => {
    let isMounted = true;

    RecordsApiV2.getAll()
      .then((response) => {
        const recs = response.data.map((item, index) => (
          <TableRow
            style={{
              backgroundColor: index % 2 === 0 ? "lightgrey" : "lightblue",
            }}
            key={index}
            item={item}
          />
        ));

        isMounted && setRecords(recs);
      })
      .catch((error) => console.log(error));

    return () => {
      isMounted = false;
    };
  }, []);

  // TODO: play with [login|logout]Redirect and other APIs
  return (
    <div className="App">
      <UnauthenticatedTemplate>
        <button onClick={() => instance.loginPopup()}>login</button>
      </UnauthenticatedTemplate>

      <AuthenticatedTemplate>
        <p>{isAuthenticated && `Hello, ${accounts[0].username}`}</p>
        <button onClick={() => instance.logoutPopup()}>logout</button>
      </AuthenticatedTemplate>
      <p>
        This is (obviously) very crude and a work in progress. The goal is to
        build a web app alongside the mobile app for the research team to better
        interact with their data and also for volunteers to further explore
        their contributions.
      </p>
      <br />
      <table>
        <TableHeader />
        <tbody>{records}</tbody>
      </table>
    </div>
  );
}

export default App;
