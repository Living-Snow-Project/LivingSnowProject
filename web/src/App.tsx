import React, { useEffect, useRef, useState } from "react";
import "./App.css";
import { serviceEndpoint } from "./constants/service";
import { TableHeader, TableRow } from "./components/TableRow";

function App() {
  const [records, setRecords] = useState("records");
  const alreadyRan = useRef(false);
  useEffect(() => {
    if (alreadyRan.current === true) {
      return;
    }

    fetch(`${serviceEndpoint}/api/records`)
      .then((response) => {
        if (!response.ok) {
          return;
        }

        console.log(response);

        response
          .json()
          .then((data) => {
            const recs = data.map((item: any, index: any) => {
              return (
                <TableRow
                  style={{
                    backgroundColor:
                      index % 2 === 0 ? "lightgrey" : "lightblue",
                  }}
                  key={item.id}
                  item={item}
                />
              );
            });
            setRecords(recs);
          })
          .catch((error) => console.log(`${error}`));
      })
      .catch((error) => setRecords(`${error}`));

    return () => {
      alreadyRan.current = true;
    };
  }, []);

  return (
    <div className="App">
      <p>
        This is (obviously) very crude and a work in progress. The goal is to
        build a web app alongside the mobile app for the research team to better
        interact with their data and also for volunteers to further explore
        their contributions.
      </p>
      <br />
      <table>
        <TableHeader />
        <tbody>{records}</tbody>
      </table>
    </div>
  );
}

export default App;
