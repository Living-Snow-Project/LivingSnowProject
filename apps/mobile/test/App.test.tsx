import * as React from "react";
import { preventAutoHideAsync } from "expo-splash-screen";
import { act, create } from "react-test-renderer";
/* eslint-disable import/extensions */
import { App } from "../App";

jest.mock("expo-splash-screen", () => {
  const actual = jest.requireActual("expo-splash-screen");
  return {
    ...actual,
    preventAutoHideAsync: jest.fn().mockReturnValue(Promise.resolve()),
  };
});

describe("App test suite", () => {
  test(`renders`, async () => {
    let tree;
    // act is used to prevent snapshot returning null
    // act is responsible for flushing all effects and rerenders after invoking it.
    await act(async () => {
      tree = await create(<App />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test("renders if SplashScreen has a failure", async () => {
    // @ts-ignore
    preventAutoHideAsync.mockImplementation(() => {
      const success = false;
      return Promise.reject(success);
    });

    let tree;
    await act(async () => {
      tree = await create(<App />);
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
