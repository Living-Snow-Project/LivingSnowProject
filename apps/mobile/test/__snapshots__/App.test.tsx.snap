// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test suite renders 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`App test suite renders if SplashScreen has a failure 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
